app/services/sso/openid_connect.rb

```diff
@@ -20,6 +20,24 @@ def authenticated?
       self.user = User.current.presence || authenticate!
     end
 
+    def default_logout_url
+      Setting['login_delegation_logout_url']
+    end
+
+    # If mod_auth_openidc is in use then User needs to logout from the
+    # OpenID provider as well as mod_auth_openidc.
+    def apache_logout_path
```

Comment:
Why is this method needed in the first place? Shouldn't this just be the `logout_url` method? That way you don't need to special case it in the users controller.

Reply:
makes sense. I will put everything into the `logout_url` method in `openid_connect.rb` file and let the user controller be untouched.